### ISSUE TITLE
Streams take way longer to open with anipy-cli v3.0.3 (up to 1 min) compared to v2.7.31 (instant)

### ISSUE THREAD

**Issue by GreatTeacherOni** [sdaqo/anipy-cli#163]
**Describe the bug**
Extraction when trying to watch streams (on gogoanime) takes very long. Compared to anipy-cli v2.7.31 which opened the stream in seconds.

**To Reproduce**
1. Type "anipy-cli" in your bash cli
2. Press Enter to skip "seasonal" and then type the name of the stream which you want to watch
4. Select the show
5. Press Enter to watch in sub
6. Select the episode
7. Wait for stream to open while it shows "Extracting streams for {name of show} + {episode number}"

OR

1. Type "anipy-cli --search "{name of show}:{episode number}:{sub/dub}"
2. Wait for stream to open while it shows "Extracting streams for {name of show} + {episode number}"

**Expected behavior**
Stream should open in seconds

**Actual behavior**
Stream can take up to 1 min to open

**Info**
Post your output of `pip show anipy-cli`:
Name: anipy-cli
Version: 3.0.3
Summary: Watch and Download anime from the comfort of your Terminal
Home-page: https://sdaqo.github.io/anipy-cli
Author: sdaqo
Author-email: sdaqo.dev@protonmail.com
License: GPL-3.0
Location: /home/moha/.cache/pypoetry/virtualenvs/anipycli-iWLBxt5Y-py3.11/lib/python3.11/site-packages
Requires: anipy-api, appdirs, inquirerpy, pypresence, pyyaml, yaspin
Required-by:

**Platform**
 - OS: NixOS 24.05
 - Default player: mpv (0.37.0)
 - Python 3.11.8
 - ffmpeg 6.1.1

**Additional context**
Updated from anipy-cli v2.7.31 to v3.0.3 and was asked to migrate config, which I did.
I guess the issue lies with the way the stream tries to open while it says "Extracting streams ...". The old way was more stable and faster.

**Comment by sdaqo** [comment]
Can you give me a example anime? The extracting logic has essentially not changed, so this should not be a problem.

**Comment by GreatTeacherOni** [comment]
> Can you give me a example anime? The extracting logic has essentially not changed, so this should not be a problem.

Example animes: Kenka Dokugaku, Kono Subarashii Sekai ni Shukufuku wo! 3

Kono Subarashii Sekai ni Shukufuku wo! 3 Episode 6 (sub) took 0m49,874s to open after typing `anipy-cli --search "Kono Subarashii Sekai ni Shukufuku wo! 3:6:sub" --ffmpeg`

**Comment by sdaqo** [comment]
Weird, this takes maybe 2 seconds for me...
Does this happen with every anime? Have you tried going on the gogoanime website and using that, is  it also that slow?
Tbh, I do not really know what is causing that, but I do not think this has something to do with the new version, it is probably the provider (gogo in this case) that is slowing things down for you, it is absolutely possible that if you try it one day later it works like normal again.

(Btw., the `--ffmpeg` option only does something when downloading.)

**Comment by GreatTeacherOni** [comment]
You're right. The website was slow for me too. Today it starts either instant or in a couple of seconds. Yeah I use --ffmpeg only so when i press "d" in the menu while the stream is active it downloads as "mp4"-file (instead of ts).

Btw., it would be nice to be asked in which format the file should be downloaded when using the internal m3u8 downloader (at least mp4 and mkv besides the default ts file format). Should I open a feature request for that? I can only see "bug report" when trying to open an issue though

**Comment by sdaqo** [comment]
> Btw., it would be nice to be asked in which format the file should be downloaded when using the internal m3u8 downloader (at least mp4 and mkv besides the default ts file format). Should I open a feature request for that? I can only see "bug report" when trying to open an issue though.

That is already a feature! you can set the `remux_to` config option.

> I also wanted to request a search feature when using the history mode via "anipy-cli -H" (similar to the way when trying to choose an episode) since it is quite time-consuming when trying to search through the whole list if the list is quite long.

Will do that!

**Comment by candrapersada** [comment]
> That is already a feature! you can set the `remux_to` config option. This also works when not using --ffmpeg, you need ffmpeg installed though.
> 
> >

remux_to: .mp4
! An error occurred during download: Error opening input files: Option not found
! 2 retries remain

v3.3.1

**Comment by S0litaire** [comment]
Same error with me when using .mkv or .MP4.
ffmpeg is installed.

I still have the .ts files so I can manually convert if I need to.

**Comment by sdaqo** [comment]
This should be fixed, but next time please open a new issue.
